library/include/internal/hipsolver-compat.h

```diff
@@ -0,0 +1,1592 @@
+/* ************************************************************************
+ * Copyright 2020-2022 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+
+/*! \file
+ *  \brief hipsolver-compat.h provides a compatibility API for users of
+ *  cuSOLVER wishing to transition to hipSOLVER. Functions in this file
+ *  have parameter lists matching those of cuSOLVER, but may suffer from
+ *  performance issues when using the rocSOLVER backend. Switching to the
+ *  use of functions from hipsolver-functions.h is recommended.
+ */
+
+#ifndef HIPSOLVER_COMPAT_H
+#define HIPSOLVER_COMPAT_H
+
+#include "hipsolver-functions.h"
+#include "hipsolver-types.h"
+
+typedef void* hipsolverDnHandle_t;
```

Comment:
I missed this before, but why do we need `hipsolverDnHandle_t`?

Reply:
The cuSOLVER handle type is called `cusolverDnHandle_t`. I figured since we're going for convenience with the compatibility API, more closely matching the handle type name would be convenient too.

Reply:
My concern is that users may assume that they need a `hipsolverDnHandle_t` to use the compatibility functions. We can explain to them that they do not, but it's better to not have to explain anything at all.

A proliferation of functions is no big deal, but a proliferation of types causes (real or perceived) incompatibilities. Given that we want people to use the original hipsolver API whenever possible, they will probably be using a mix of both APIs in the same code. The fact that the types don't match may be a bit confusing.

Reply:
That is a good point!
And in that case, we may also eliminate the Dn create, destroy, setstream and getstream helpers. In other words, the compat API will contain only functionality APIs.

Reply:
I don't mind removing the handle type, but I'm a little more dubious when it comes to removing functions (even helper functions).

Actually, would it make more sense to define `hipsolverDnHandle_t` in terms of `hipsolverHandle_t`? That way, the connection will be obvious.

Reply:
> would it make more sense to define `hipsolverDnHandle_t` in terms of `hipsolverHandle_t`? That way, the connection will be obvious.

Maybe we could do that, but use `hipsolverHandle_t` in all our API signatures? Then it becomes very obvious you can use `hipsolverHandle_t` everywhere, but you can still port your code with `s/hipsolver/cusolver/g`.

Reply:
Sounds good. Let's do that.